File: 3736.minimum-moves-to-equal-array-elements-iii.py
#
# @lc app=leetcode id=3736 lang=python
#
# [3736] Minimum Moves to Equal Array Elements III
#

# @lc code=start
class Solution(object):
    def minMoves(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        m = - float('inf')
        s = 0
        for i in nums:
            s += i 
            m = max(m,i)
        return m*len(nums) - s
        
# @lc code=end

File: 3736.minimum-moves-to-equal-array-elements-iii.java
/*
 * @lc app=leetcode id=3736 lang=java
 *
 * [3736] Minimum Moves to Equal Array Elements III
 */

// @lc code=start
class Solution {
    public int minMoves(int[] nums) {
        int ma = Integer.MIN_VALUE;
        int su = 0;
        for (int i : nums) {
            ma = Math.max(ma, i);
            su += i;
        }
        return (ma * nums.length) - su;
    }
}
// @lc code=end

